data-pipeline/src/telemetry/mod.rs

```diff
@@ -0,0 +1,418 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod metrics;
+use crate::telemetry::metrics::Metrics;
+use anyhow::Result;
+use ddcommon::tag;
+use ddcommon::tag::Tag;
+use ddtelemetry::data::metrics::{MetricNamespace, MetricType};
+use ddtelemetry::metrics::ContextKey;
+use ddtelemetry::worker::{
+    LifecycleAction, TelemetryActions, TelemetryWorkerBuilder, TelemetryWorkerHandle,
+};
+use std::collections::HashMap;
+use std::time::Duration;
+use tokio::select;
+use tokio::task::JoinHandle;
+
+#[derive(Debug, PartialEq)]
+struct SelfMetric(MetricType, MetricNamespace);
+
+/// Structure to build a Telemetry client.
+///
+/// It will hold partial data until `build` method is called which result in a new
+/// `TelemetryClient`.
+#[derive(Default)]
+pub struct TelemetryClientBuilder {
+    service_name: Option<String>,
+    language: Option<String>,
+    language_version: Option<String>,
+    tracer_version: Option<String>,
+    metrics: HashMap<String, SelfMetric>,
+    heartbeat: u64,
+    interval: u64,
+    url: Option<String>,
+}
+
+impl TelemetryClientBuilder {
+    /// Creates a new empty builder.
+    pub fn new() -> Self {
```

Comment:
I don't feel strongly about this, but if we're just returning a default `Self`, do we need a `new()` method at all?